File: test_api.py
#!/usr/bin/env python3
"""
Simple script to test the Paper Trading API endpoints.
Run this after starting the API server with `python run_api.py`.
"""

import requests
import json
from datetime import datetime, timedelta

BASE_URL = "http://localhost:8000"

def print_response(response, endpoint):
    """Print formatted API response"""
    print(f"\n=== {endpoint} ===")
    print(f"Status: {response.status_code}")
    if response.status_code == 200:
        try:
            print(json.dumps(response.json(), indent=2))
        except:
            print(response.text)
    else:
        print(f"Error: {response.text}")
    print("=" * 50)

def test_api():
    """Test various API endpoints"""
    
    # Test root endpoint
    response = requests.get(f"{BASE_URL}/")
    print_response(response, "Root Endpoint")
    
    # Test account endpoints
    response = requests.get(f"{BASE_URL}/account/balance")
    print_response(response, "Account Balance")
    
    response = requests.get(f"{BASE_URL}/account/usdt-balance")
    print_response(response, "USDT Balance")
    
    response = requests.get(f"{BASE_URL}/account/portfolio")
    print_response(response, "Portfolio")
    
    # Test market endpoints
    response = requests.get(f"{BASE_URL}/market/pairs")
    print_response(response, "Currency Pairs")
    
    response = requests.get(f"{BASE_URL}/market/price/BTC%2FUSDT")
    print_response(response, "BTC/USDT Price")
    
    # Test order endpoints
    response = requests.get(f"{BASE_URL}/orders/fulfilled")
    print_response(response, "Fulfilled Orders")
    
    response = requests.get(f"{BASE_URL}/orders/open")
    print_response(response, "Open Orders")
    
    # Test PnL calculation
    now = datetime.now()
    one_year_ago = now - timedelta(days=365)
    
    time_range = {
        "start_time": one_year_ago.strftime("%Y-%m-%d %H:%M:%S"),
        "end_time": now.strftime("%Y-%m-%d %H:%M:%S")
    }
    
    response = requests.post(
        f"{BASE_URL}/trades/pnl/calculate", 
        json=time_range
    )
    print_response(response, "PnL Calculation")

if __name__ == "__main__":
    print("Testing Paper Trading API...")
    test_api()
    print("\nAPI testing completed!")

File: trading-ui/test-api.js
// Simple test script to verify API endpoints
const API_BASE_URL = 'http://localhost:8001';

async function testEndpoint(endpoint, description) {
  try {
    console.log(`\n🧪 Testing: ${description}`);
    console.log(`📡 Endpoint: ${endpoint}`);
    
    const response = await fetch(`${API_BASE_URL}${endpoint}`);
    
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }
    
    const data = await response.json();
    console.log(`✅ Success: ${JSON.stringify(data).substring(0, 100)}...`);
    return data;
  } catch (error) {
    console.log(`❌ Error: ${error.message}`);
    return null;
  }
}

async function testPOSTEndpoint(endpoint, body, description) {
  try {
    console.log(`\n🧪 Testing: ${description}`);
    console.log(`📡 Endpoint: ${endpoint}`);
    console.log(`📦 Body: ${JSON.stringify(body)}`);
    
    const response = await fetch(`${API_BASE_URL}${endpoint}`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(body),
    });
    
    if (!response.ok) {
      throw new Error(`HTTP ${response.status}: ${response.statusText}`);
    }
    
    const data = await response.json();
    console.log(`✅ Success: ${JSON.stringify(data).substring(0, 100)}...`);
    return data;
  } catch (error) {
    console.log(`❌ Error: ${error.message}`);
    return null;
  }
}

async function runTests() {
  console.log('🚀 Starting API Integration Tests');
  console.log('=' .repeat(50));
  
  // Test all the endpoints our UI will use
  await testEndpoint('/account/usdt-balance', 'USDT Balance');
  await testEndpoint('/account/portfolio', 'Portfolio Holdings');
  await testEndpoint('/trades/history', 'Trade History');
  
  // Test PnL calculation
  await testPOSTEndpoint('/trades/pnl/calculate', {
    start_time: '2025-05-25 12:00:00',
    end_time: '2025-06-07 23:59:59'
  }, 'PnL Calculation');
  
  console.log('\n🏁 Tests completed!');
}

// Run the tests
runTests();
